src/network/clientpackethandler.cpp

```diff
@@ -968,6 +969,29 @@ void Client::handleCommand_SpawnParticle(NetworkPacket* pkt)
 	m_client_event_queue.push(event);
 }
 
+void Client::handleCommand_SpawnParticlesBatched(NetworkPacket *pkt)
+{
+	std::string decompressed_str;
+	{
+		std::istringstream iss(pkt->readLongString(), std::ios::binary);
+		std::ostringstream oss;
+		decompressZstd(iss, oss);
+		decompressed_str = oss.str();
+	}
+	std::istringstream iss(decompressed_str, std::ios::binary);
```

Comment:
untested. Should work, though.
```suggestion
	std::istringstream iss;
	{
		std::istringstream iss(pkt->readLongString(), std::ios::binary);
		std::ostringstream oss;
		decompressZstd(iss, oss);
		iss = std::istringstream(oss.str(), std::ios::binary);
	}
```

Reply:
> EDIT: See also: Client::handleCommand_NodeDef

Okay, I see what you mean. Switched both to in+out stringstreams. (I just went with `particle_data` to put emphasis on what's inside rather than it being first written to and then read from.)